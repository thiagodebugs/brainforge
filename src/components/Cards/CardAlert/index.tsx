import Image from "next/image";
import Link from "next/link";
import style from "./style.module.css";

interface CardAlertProps {
  cardAlert: {
    title: string;
    description: string;
    image: string;
    link: string;
  };
}

export default function CardAlert({ cardAlert }: CardAlertProps) {
  return (
    <>
      <Link href={cardAlert.link} className={style.cardAlert}>
        <Image
          src={cardAlert.image}
          alt={cardAlert.title}
          width={1080}
          height={1080}
        />
        <div className={style.cardAlertTitle}>
          <h1>{cardAlert.title}</h1>
          <p>{cardAlert.description}</p>
        </div>
      </Link>
    </>
  );
}
