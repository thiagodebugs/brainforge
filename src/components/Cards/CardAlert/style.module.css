.cardAlert {
  @apply flex flex-row items-center bg-neutral-800 rounded-xl transition-all duration-300 transform border-[#26c666] hover:scale-105 hover:border-2 hover:border-[#26c666] ease-in-out w-full h-24 overflow-hidden;
}

.cardAlert img {
  @apply max-w-40 h-full object-cover;
}

.cardAlertTitle {
  @apply space-y-2 px-6;
}

/* ocultar resto para baixo */
.cardAlertTitle h1 {
  @apply font-semibold text-sm lg:text-base overflow-hidden max-h-10 lg:max-h-12;
}

.cardAlertTitle p {
  @apply text-xs font-extralight hidden lg:block whitespace-nowrap overflow-hidden text-ellipsis max-w-sm;
}
