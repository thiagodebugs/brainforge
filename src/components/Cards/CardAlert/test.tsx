import { render } from "@testing-library/react";
import CardAlert from ".";

const props = {
  title: "Lorem ipsum",
  description: "Test",
  image: "/images/monthyly.png",
  link: "/test",
};

describe("CardAlert", () => {
  it("should render the card alert", () => {
    const { container } = render(<CardAlert cardAlert={props} />);

    expect(container).toMatchSnapshot();
  });

  it("click on card alert", () => {
    const { getByText } = render(<CardAlert cardAlert={props} />);

    expect(getByText("Lorem ipsum")).toBeInTheDocument();
    expect(getByText("Test")).toBeInTheDocument();
  });
});
