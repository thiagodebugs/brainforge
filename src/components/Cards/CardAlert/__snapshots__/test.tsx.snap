// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardAlert should render the card alert 1`] = `
<div>
  <a
    class="cardAlert"
    href="/test"
  >
    <img
      alt="Lorem ipsum"
      data-nimg="1"
      decoding="async"
      height="1080"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fmonthyly.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Fmonthyly.png&w=1080&q=75 1x, /_next/image?url=%2Fimages%2Fmonthyly.png&w=3840&q=75 2x"
      style="color: transparent;"
      width="1080"
    />
    <div
      class="cardAlertTitle"
    >
      <h1>
        Lorem ipsum
      </h1>
      <p>
        Test
      </p>
    </div>
  </a>
</div>
`;
